sqldelight-gradle-plugin/src/main/kotlin/com/squareup/sqldelight/gradle/VerifyMigrationTask.kt

```diff
@@ -36,7 +36,7 @@ open class VerifyMigrationTask : SourceTask() {
   @TaskAction
   fun verifyMigrations() {
     // Clear existing build directory.
-    File("${project.buildDir}/sqldelight").deleteRecursively()
+    File("${project.buildDir}/sqldelight-verifyMigration").deleteRecursively()
```

Comment:
I'm not sure this is enough. The Android and/or Kotlin multiplatform support likely means there are multiple of these tasks per module.

We probably need to move responsibility of creating this directory to the person who is setting up this task.

Additionally, I would prefer if we put everything under the 'sqldelight' folder instead of having multiple.

Reply:
Good point. I'd be happy to try and change the plugin to use directories something like this:
* generateInterface tasks: `build/sqldelight/generated/{variant}`
* verifyMigration tasks: `build/sqldelight/verifymigration/{variant}`

Are there any compatibility concerns with changing the current `build/sqldelight` output directory?

Also, feel free to suggest better names. I haven't done much with Gradle, so don't know the preferred conventions.

Note that variants aren't really working too well generally anyway, because the generateInterface tasks all delete and write to `build/sqldelight` and clobber each other. So for example in an Android project generateDebugDatabaseInterface and generateReleaseInterface will race with each other.

Reply:
No concerns changing directory structure. No one should be relying on it, and if they are we need to expose API for it, not just having it be relied on implicitly. Your names seem fine, though.

Reply:
i think we can pretty safely make the change to have the current generate tasks output to `build/sqldelight/generated` and then have verification tasks output to `build/sqldelight/verifyMigration`

we currently dont do output to any variant specific folders. It's not the most lovely setup if you peak under the hood but it works, I don't think we need to change that in this PR but can investigate further later.

Reply:
We also need to have the database name in the path.

I think we want to do something like `build/generated/sqldelight/REASON/DATABASE_NAME/VARIANT/` where `REASON` is "code" or "migration_verification", `DATABASE_NAME` is the DB name (obviously), and `VARIANT` is the android and/or kotlin MPP variant.